tasks/helpers/bsc-fork.ts

```diff
@@ -0,0 +1,40 @@
+import { ethers } from 'ethers';
+import { NETWORKS_RPC_URL } from '../../helper-hardhat-config';
+import { DRE, getNetworkName, isForkNetwork, sleep } from '../../helpers/misc-utils';
+import { eOtherNetwork } from '../../helpers/types';
+
+export const workaroundForBSCFork = async () => {
+  if (!isForkNetwork() || getNetworkName() != eOtherNetwork.bsc) {
+    return;
+  }
+
+  const url = NETWORKS_RPC_URL[eOtherNetwork.bsc];
+  const bscProvider = new ethers.providers.JsonRpcProvider(url);
+  let latestBlock = (await bscProvider.getBlockNumber()) + 5;
+
+  console.log('WARNING! Resetting HardHat to fork BSC at ', latestBlock);
+  console.log('===========================================================\n');
+
+  while (true) {
+    try {
+      await DRE.network.provider.send('hardhat_reset', [
+        {
+          forking: {
+            jsonRpcUrl: url,
+            blockNumber: latestBlock,
+          },
+        },
+      ]);
+      return;
+    } catch (err: any) {
```

Comment:
JFYI: it's possible to use global type Error.

Reply:
sometimes there isn't Error at all